src/couch/entry.js

```diff
@@ -234,21 +234,15 @@ const methods = {
         });
         result = await updateEntry(this, doc, entry, user, options);
       } catch (e) {
-        if (e.reason === 'not found') {
-          result = await notFound(e);
-          action = 'created';
-        } else {
-          throw e;
-        }
+        await createIfNotFound(e);
       }
     } else if (entry.$id) {
       debug.trace('entry has no _id but has $id: %o', entry.$id);
-      if (await getUniqueEntryById(this, user, entry.$id)) {
-        throw new CouchError('entry already exists', 'conflict');
-      } else {
-        result = await notFound({ reason: 'not found' });
-        action = 'created';
+      if (options.isUpdate) {
+        throw new CouchError('Document does not exist', 'not found');
       }
+      result = await createNew(this, entry, user, options);
```

Comment:
Is there a risk that a previous call never finishes and ends up in a deadlock?

Reply:
I checked and yes there was because got has no timeout by default. All the asynchronous work is done in requests managed by got so I think there are no more risks now that I added a timeout of 60s